ravenframework/SupervisedLearning/DMD/HODMD.py

```diff
@@ -0,0 +1,197 @@
+    specs.addSub(InputData.parameterInputFactory("reconstruction_method", contentType=InputTypes.makeEnumType("reconstruction_method", "ReconstructionMethodType",
+                                                                                                        ["first", "mean"]),
+                                                 descr=r"""Method used to reconstruct the snapshots of the dynamical system from the multiple versions available
+                                                 due to how HankelDMD is conceived. If ``first'' (default) the first version available is selected
+                                                 (i.e. the nearest to the 0-th row in the augmented matrix). If ``mean'' we compute the element-wise mean.
+                                                 """, default='first'))
+    return specs
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the distribution parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    import pydmd
+    from pydmd import HODMD
+    super()._handleInput(paramInput)
+    settings, notFound = paramInput.findNodesAndExtractValues(['svd_rank', 'tlsq_rank','exact', 'opt', 'rescale_mode','svd_rank_extra',
+                                                               'forward_backward', 'd', 'sorted_eigs', 'reconstruction_method'])
+    # notFound must be empty
+    assert(not notFound)
+    # -1 no truncation, 0 optimal rank is computed, >1 truncation rank
+    # if 0.0 < float < 1.0, computed rank is the number of the biggest sv needed to reach the energy identified by this float value
+    self.dmdParams['svd_rank'] = settings.get('svd_rank')
+    # truncation rank for total least square
+    self.dmdParams['tlsq_rank'] = settings.get('tlsq_rank')
+    # True if the exact modes need to be computed (eigs and eigvs), otherwise the projected ones (using the left-singular matrix)
+    self.dmdParams['exact'      ] = settings.get('exact')
+    # amplitudes computed minimizing the error between the mods and all the timesteps (True) or 1st timestep only (False)
+    self.dmdParams['opt'] = settings.get('opt')
+    # Rescale mode
+    self.dmdParams['rescale_mode'] = settings.get('rescale_mode')
+    if self.dmdParams["rescale_mode"] == 'None':
+      self.dmdParams["rescale_mode"] = None
+    # Sorted eigs
+    self.dmdParams['sorted_eigs'] = settings.get('sorted_eigs')
+    if self.dmdParams["sorted_eigs"] == 'False':
+      self.dmdParams["sorted_eigs"] = False
+    # Forward Backward method (see FbDMD)
+    self.dmdParams['forward_backward'] = settings.get('forward_backward')
+    # Sorted eigs
+    self.dmdParams['d'] = settings.get('d')
+    # Reconstruction method
+    self.dmdParams['reconstruction_method'] = settings.get('reconstruction_method')
+    # svd_rank_extra
+    self.dmdParams['svd_rank_extra'] = settings.get('svd_rank_extra')
+
```

Comment:
I think you can use a for loop here to update self.dmdParams here. This suggestion will apply to other classes as well.